types/rdf-js/index.d.ts

```diff
@@ -307,7 +307,7 @@ export interface DataFactory<Q extends BaseQuad = Quad> {
      * @return A new instance of Quad.
      * @see Quad
      */
-    quad(subject: Q['subject'], predicate: Q['predicate'], object: Q['object'], graph?: Q['graph']): Q;
+    quad(subject: InQuad['subject'], predicate: InQuad['predicate'], object: InQuad['object'], graph?: InQuad['graph']): OutQuad;
```

Comment:
@rubensworks @LaurensRietveld 

While improving `rdf-ext` I noticed that the factory might require similar treatment that the `DatasetCore` got before. By splitting the type arguments I was able to accurately type `DataFactoryExt` which can take any base term and use them to build it's sepcialized `QuadExt`